platform/platform.go

```diff
@@ -184,6 +194,21 @@ type sshPipe struct {
 	io.Reader
 }
 
+func init() {
+	// Determine endianness
```

Comment:
I'd add a link to the stackoverflow post to credit it.

Not sure if `platform.go` is the most elegant place to put this but I don't know the code well enough to suggest somethign else.

Reply:
done. `platform.go` looked like a good place to put a basic platform specific value like endianness. But again, I am new to this project too so will let @arithx comment.

Reply:
With the new implementation I'd probably lean towards having it in `util`. It doesn't really have anything to do with the platform that we're trying to launch but rather is determining something about the host machine, which happens to affects how we need to look for the port number in the `qemu-unpriv` platform.

Reply:
OK. in that case i will put it in `util/common.go`.So in that case even the machine specific properties like architecture which are needed to make decisions on selection will be in util in the new implementation?

Reply:
Elaborating a little bit further, to me the `util` package isn't the perfect place for it (ideally `go` would have a built-in) but is probably the one that makes the most sense to me.

> So in that case even the machine specific properties like architecture which are needed to make decisions on selection will be in util in the new implementation?

I'm not sure I understand, this isn't a property of the machine being tested but rather the machine that the tests are being ran from. Properties of the machine under test are currently (and will still be) stored inside of the `Options` variable that is passed to the `Flight`.

For example you mentioned `architecture` in your post. There are two different architectures at play here, the host architecture (machine that `kola` is being ran from) and the target architecture (machine under test; currently the `board` variable). The host architecture is only relevant because there are different QEMU commands for different architectures and we need to know which one is available on the system and is currently determined by looking at the `runtime.GOARCH` built-in. The target architecture is a property of the test that we want to run, it can have implications on not only how we might need to spawn the test (e.x.: in `qemu` we need to specify the `cpu` type) but could also impact what tests are ran or how we pull information from the machine.

Does that answer your question?

Reply:
Ah...my understanding was off..I assumed that host=target arch. Thank you for the detailed explanation. Makes things much clearer.